kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/codec/ResponseDecoderTest.java

```diff
@@ -126,4 +129,22 @@ private static ByteBuf toLength5ByteBuf(int i) {
         });
     }
 
+    @Test
+    void shouldNotThrowIfApiVersionsResponseUnsupported() {
+        mgr.putBrokerRequest(ApiKeys.API_VERSIONS.id, (short) 3, 52, true, null, null, true);
+
+        // given
+        ByteBuf buffer = Unpooled.wrappedBuffer(serializeUsingKafkaApis((short) 0,
+                exampleResponseHeader(),
+                (short) 0,
+                new ApiVersionsResponseData()
+                        .setErrorCode(Errors.UNSUPPORTED_VERSION.code())));
+        ArrayList<Object> objects = new ArrayList<>();
+
+        // when
+        responseDecoder.decode(null, buffer, objects);
+
+        // then
+        assertThat(objects).hasSize(1);
```

Comment:
how about strengthening the assertion to check the decoded frame?

```suggestion
        assertThat(objects)
                .singleElement()
                .extracting("body")
                .isEqualTo(new ApiVersionsResponseData().setErrorCode(Errors.UNSUPPORTED_VERSION.code()));
```